lib/routes/utils/oauth-mint.js

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const jwtool = require('fxa-jwtool')
+const Pool = require('../../pool')
+const rsa = require('../../crypto/rsa')
+
+// From the Firefox source code:
+// Having this parameter be short has limited security value and can cause
+// spurious authentication values if the client's clock is skewed and
+// we fail to adjust. See Bug 983256.
+const ASSERTION_LIFETIME = 1000 * 3600 * 24 * 365 * 25 // 25 years in ms.
+const CERT_LIFETIME = 1000 * 3600 * 5 // 5 hours in ms.
+
+module.exports = (log, config, signer) => {
+  const SafeUrl = require('../../safe-url')(log)
+  const path = new SafeUrl('/v1/authorization')
+  const keysPromise = rsa.generateKeypair().then(({prv, pub}) => {
```

Comment:
As reassurance to future readers, please add a comment here that this is shared at the module-level on purpose, along the lines of:  "To avoid an expensive key-generation for every request, we use a single shared keypair generated at startup.".